### PR TITLE
Add callback function before/after the initial Esolve

### PR DESCRIPTION
**Pull request by RemiLehe**
This adds new callback functions:
- `beforeInitEsolve`
- `afterInitEsolve`

that are called before/after the Poisson solve that we do before the beginning of the PIC loop.

### CODE REVIEW

**Review by dpgrote — APPROVED** [review]
Looks good. Thanks!

**Review by roelof-groenewald — DISMISSED** [review]

File: Source/Initialization/WarpXInitData.cpp [hunk 1]
```diff
@@ -413,7 +414,9 @@ WarpX::InitData ()
     {
         // Loop through species and calculate their space-charge field
         bool const reset_fields = false; // Do not erase previous user-specified values on the grid
+        ExecutePythonCallback("beforeEsolve");
```
Comment (RemiLehe) [inline Source/Initialization/WarpXInitData.cpp]:
```suggestion
        ExecutePythonCallback("beforeInitEsolve");
```

File: Source/Initialization/WarpXInitData.cpp [hunk 2]
```diff
@@ -413,7 +414,9 @@ WarpX::InitData ()
...
         ComputeSpaceChargeField(reset_fields);
+        ExecutePythonCallback("afterEsolve");
```
Comment (RemiLehe) [inline Source/Initialization/WarpXInitData.cpp]:
```suggestion
        ExecutePythonCallback("afterInitEsolve");
```

File: Python/pywarpx/callbacks.py [hunk 3]
```diff
@@ -352,13 +385,13 @@ def callfrombeforeEsolve(f):
     installbeforeEsolve(f)
     return f
 def installbeforeEsolve(f):
-    "Adds a function to the list of functions called before an E solve"
+    "Adds a function to the list of functions called before the initial E solve"
```
Comment (dpgrote) [inline Python/pywarpx/callbacks.py]: Were these changes inadvertent?

**Comment by RemiLehe** [comment]
Thanks for the quick feedback!
I corrected the accidental changes in the docstring of `afterEsolve`, and updated `WarpX_py.H`. Do you think that this is ready for merging?

**Review by dpgrote — APPROVED** [review]
Looks good. Thanks!

**Review by roelof-groenewald — APPROVED** [review]
Thanks!

**Pull request merged**